### PR TITLE
Adding trace to reflector initialization

### PR DESCRIPTION
**Pull request by krzysied**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This change allows to spot and investigate long (> 10s) reflector initialization.

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/cache/reflector.go [hunk 1]
```diff
@@ -174,26 +175,33 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
...
 	}
+	trace.Step("Objects extracted")
 	r.metrics.numberOfItemsInList.Observe(float64(len(items)))
 	if err := r.syncWith(items, resourceVersion); err != nil {
 		return fmt.Errorf("%s: Unable to sync list result: %v", r.name, err)
 	}
+	trace.Step("SyncWith done")
 	r.setLastSyncResourceVersion(resourceVersion)
+	trace.Step("Resource version updated")
+	trace.LogIfLong(10 * time.Second)
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/tools/cache/reflector.go]: This won't work if there will be error returned.
We should be doing defere trace.LogIfLong.

If you want to finish it here, you can pack this whole block into:
```
func() {
// here what there is now
}()
```

**Review by neolit123 — COMMENTED** [review]
/priority important-longterm

**Comment by liggitt** [comment]
this PR made k8s.io/client-go depend on k8s.io/apiserver, which is not desired. please rework to remove that dependency and restore the import restriction that disallowed that

**Comment by krzysied** [comment]
@liggitt @wojtek-t Sure, I will fix. I will create PR today.

File: hack/import-restrictions.yaml [hunk 2]
```diff
@@ -82,6 +83,7 @@
   - "./vendor/k8s.io/client-go/tools/remotecommand"
   allowedImports:
   - k8s.io/apimachinery
+  - k8s.io/apiserver/pkg/util/trace
```
Comment (sttts) [inline hack/import-restrictions.yaml]: this is forbidden. Please revert. We cannot depend on eachother.

Comment (nikhita) [reply hack/import-restrictions.yaml]: https://github.com/kubernetes/kubernetes/pull/73209 :)

**Pull request merged**